src/ORM/Search/BasicSearchContext.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace SilverStripe\ORM\Search;
+
+use BadMethodCallException;
+use InvalidArgumentException;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Dev\Deprecation;
+use SilverStripe\ORM\Filterable;
+use SilverStripe\ORM\Limitable;
+use SilverStripe\ORM\Sortable;
+
+/**
+ * A SearchContext that can be used with non-ORM data.
+ */
+class BasicSearchContext extends SearchContext
+{
+    use Configurable;
+
+    /**
+     * Name of the field which, if included in search forms passed to this object, will be used
+     * to search across all searchable fields.
+     */
+    private static $general_search_field_name = 'q';
+
+    /**
+     * Returns a list which has been limited, sorted, and filtered by the given parameters.
+     *
+     * @param array $searchParams Map of search criteria, mostly taken from $_REQUEST.
+     *  If a filter is applied to a relationship in dot notation,
+     *  the parameter name should have the dots replaced with double underscores,
+     *  for example "Comments__Name" instead of the filter name "Comments.Name".
+     * @param array|bool|string $sort Field to sort on.
+     * @param array|null|string $limit
+     * @param Filterable&Sortable&Limitable $existingQuery
+     */
+    public function getQuery($searchParams, $sort = false, $limit = false, $existingQuery = null): Filterable&Sortable&Limitable
+    {
+        if (!$existingQuery || !($existingQuery instanceof Filterable) || !($existingQuery instanceof Sortable) || !($existingQuery instanceof Limitable)) {
+            throw new InvalidArgumentException('getQuery requires a pre-existing filterable/sortable/limitable list (usually ArrayList) to be passed as $existingQuery.');
+        }
+
+        if ((count(func_get_args()) >= 3) && (!in_array(gettype($limit), ['array', 'NULL', 'string']))) {
+            Deprecation::notice(
+                '5.1.0',
+                '$limit should be type of array|string|null'
+            );
+            $limit = null;
+        }
```

Comment:
This is necessary because we have to match the method signature to `SearchContext::getQuery()` which has the deprecated type in it.